sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -630,23 +647,21 @@ private BasicDocumentRevision createDocument(SQLDatabase db,String docId, final
                 options.docNumericId = docNumericId;
                 options.parentSequence = -1l;
             }
-
             options.deleted = false;
             options.current = true;
             options.data = body.asBytes();
             options.available = true;
             options.copyAttachments = false;
-            long newSequence = insertRevision(db,options);
-            if (newSequence < 0) {
-                throw new IllegalStateException("Error inserting new revision: "+options);
-            }
-
-            BasicDocumentRevision doc = getDocumentInQueue(db, docId, options.revId);
-            documentCreated = new DocumentCreated(doc);
```

Comment:
I guess you dropped `documentCreated = new DocumentCreated(doc);` because it was an unused variable. This is supposed to be posted on the event queue and must have been deleted in a previous commit. A bit bad that this wasn't caught by a unit test.